### PR TITLE
Handle redefinition of private interface methods

### PR DESCRIPTION
**Pull request by gacholio**
Correctly look up the method when doing the redefinition check for
private interface methods. Also correctly ignore final and virtual
Object methods in interface calls (Object can currently not be
redefined).

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
This case is not tested, so basic HCR regression testing should be sufficient.

File: runtime/util/hshelp.c [hunk 1]
```diff
@@ -1726,17 +1722,48 @@ fixRAMConstantPoolForFastHCR(J9ConstantPool *ramConstantPool, J9HashTable *class
 									methodResult = hashTableFind(methodHashTable, &methodPair);
 									if (NULL != methodResult) {
 										UDATA argCount = (methodRef->methodIndexAndArgCount & 255);
-										UDATA newMethodIndex = getITableIndexForMethod(methodResult->newMethod, resolvedClass);
+										J9Method *newMethod = methodResult->newMethod;
+										UDATA newMethodIndex = newMethod - J9_CLASS_FROM_METHOD(newMethod)->ramMethods;
 										/* Fix the index in the resolved CP entry, retaining the argCount */
 										methodRef->methodIndexAndArgCount = ((newMethodIndex << J9_ITABLE_INDEX_SHIFT) | argCount);
```
Comment (DanHeidinga) [inline runtime/util/hshelp.c]: Does this need the `J9_ITABLE_INDEX_METHOD_INDEX` or'd into the value ?

Comment (gacholio) [reply runtime/util/hshelp.c]: Yes it does.

**Comment by gacholio** [comment]
jenkins test sanity xlinux jdk8

**Comment by pshipton** [comment]
Starting PR testing on zlinux where machines are available
jenkins test sanity zlinux jdk8

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdk8

**Comment by pshipton** [comment]
jenkins test extended zlinux jdk11

**Pull request merged**